app/models/metric/ci_mixin/capture.rb

```diff
@@ -32,27 +46,22 @@ def perf_capture_queue(interval_name, options = {})
 
     log_target = "#{self.class.name} name: [#{name}], id: [#{id}]"
 
-    # Determine what items we should be queuing up
-    items = []
     cb = nil
     if interval_name == 'historical'
       start_time = Metric::Capture.historical_start_time if start_time.nil?
       end_time = Time.now.utc if end_time.nil?
-
-      start_hour = start_time
-      while start_hour != end_time
-        end_hour = start_hour + 1.day
-        end_hour = end_time if end_hour > end_time
-        items.unshift([interval_name, start_hour, end_hour])
-        start_hour = end_hour
-      end
     else
-      items << [interval_name]
-      items[0] << start_time << end_time unless start_time.nil?
-
+      start_time = last_perf_capture_on unless start_time
```

Comment:
The side effect of this is that it's still possible to call `target.perf_capture("realtime")` on a target with a very old `last_perf_capture_on` and end up with the bad behavior of attempting to grab many many days of metrics in one go.

This change *only* helps if the caller uses the queue to queue up the perf capture.

This leads me to think that we need to refactor the metrics capture system past the `perf_capture_queue` method to always insist that the `start_date` and `end_date` are provided.  That way, either you know what you're doing from the caller's standpoint, or, you queue up the call and let the queuing mechanism handle the break down of date intervals.

In fact, even trying to catch the problem at the `perf_capture` level leaves the individual provider implementations with the ability to interpret what it means to have a missing `start_date` or `end_date`.  It feels very dirty allowing each layer of the metrics capture code make those assumptions.